source/initial_temperature/adiabatic.cc

```diff
@@ -33,17 +33,48 @@ namespace aspect
 {
   namespace InitialTemperature
   {
+
+    template <int dim>
+    Adiabatic<dim>::Adiabatic ()
+      :
+      surface_boundary_id(numbers::invalid_unsigned_int)
+    {}
+
+    template <int dim>
+    void
+    Adiabatic<dim>::initialize ()
+    {
+      // Find the boundary indicator that represents the surface
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+      std::set<types::boundary_id> surface_boundary_set;
+      surface_boundary_set.insert(surface_boundary_id);
+
+      // The input ascii table contains one data column (LAB depths(m)) in addition to the coordinate columns.
+      Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,
+                                                    1);
+    }
+
     template <int dim>
     double
     Adiabatic<dim>::
     initial_temperature (const Point<dim> &position) const
     {
-      // convert input ages to seconds
-      const double age_top =    (this->convert_output_to_years() ? age_top_boundary_layer * year_in_seconds
-                                 : age_top_boundary_layer);
+      double age_top = 0;
       const double age_bottom = (this->convert_output_to_years() ? age_bottom_boundary_layer * year_in_seconds
                                  : age_bottom_boundary_layer);
-
+      if (read_from_ascii_file)
+        {
+          // The input ascii contains the age of the seafloor. User must provide ages in seconds
+          age_top = Utilities::AsciiDataBoundary<dim>::get_data_component(surface_boundary_id,
+                                                                          position,
+                                                                          0);
+        }
+      else
+        {
+          // convert input ages to seconds
+          age_top =    (this->convert_output_to_years() ? age_top_boundary_layer * year_in_seconds
```

Comment:
```suggestion
          age_top = (this->convert_output_to_years() ? age_top_boundary_layer * year_in_seconds
```

Reply:
Adjust line below to match ...

Reply:
Yes when implementing 

`      if (read_from_ascii_file)
        Utilities::AsciiDataBoundary<dim>::initialize(surface_boundary_set,` 

I still get an error message